lib/Dialect/HW/PortConverter.cpp

```diff
@@ -164,8 +173,14 @@ LogicalResult PortConverterImpl::run() {
     body->eraseArguments([&ports](BlockArgument arg) {
       return arg.getArgNumber() < ports.sizeInputs();
     });
-    // Set the new operands, overwriting the old ones.
-    body->getTerminator()->setOperands(newOutputValues);
+    // Set the new operands, overwriting the old ones - these are gathered from
+    // the set of temporary output ops.
+    llvm::SmallVector<Value> outputOperands;
+    llvm::transform(newOutputValues, std::back_inserter(outputOperands),
+                    [](Operation *op) { return op->getOperand(0); });
+    body->getTerminator()->setOperands(outputOperands);
+    for (auto *op : llvm::make_early_inc_range(newOutputValues))
```

Comment:
I don't think the make_early_range is necessary as you are not modifying the vector itself.